staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go

```diff
@@ -162,7 +161,12 @@ func (u *Unstructured) GetOwnerReferences() []metav1.OwnerReference {
 func (u *Unstructured) SetOwnerReferences(references []metav1.OwnerReference) {
 	newReferences := make([]interface{}, 0, len(references))
 	for _, reference := range references {
-		newReferences = append(newReferences, setOwnerReference(reference))
+		out, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&reference)
```

Comment:
This is consistent with how method `SetInitializers()` (line 357) works, future-proof against new fields and takes into account all json tags properly (see bug with pointers #46817).